frontend/pages/collections/[slug].page.tsx

```diff
@@ -78,3 +103,113 @@ export const getStaticPaths: GetStaticPaths = async () => {
     fallback: "blocking",
   };
 };
+
+const MobileViewComponent: FC<CollectionProps> = ({ collection }: CollectionProps) => {
+  const [showIcons, setShowIcons] = useState(false);
+  const onClick = () => {
+    setShowIcons(true);
+    if (showIcons) {
+      setShowIcons(false);
+    }
+  };
+  return (
+    <>
+      <Container>
+        <Grid sx={{ m: 3, display: "flex", alignItems: "center", justifyContent: "flex-end" }}>
+          <Grid sx={{ maxWidth: 250, alignItems: "center", justifyContent: "center" }}>
+            <Grid
+              sx={{
+                flexDirection: "column",
+                display: "flex",
+                alignItems: "center",
+                justifyContent: "center",
+              }}
+            >
+              <Button
+                variant="contained"
+                color="primary"
+                onClick={onClick}
+                sx={{ borderRadius: "50%" }}
+              >
+                <ShareIcon />
+              </Button>
+              <>
+                {showIcons ? (
+                  <Grid
+                    container
+                    spacing={2}
+                    alignItems="center"
+                    justifyContent="center"
+                    sx={{ my: 1 }}
+                  >
+                    <Grid item>
+                      <FacebookShareButton
+                        url={`https://modularhistory.com/collections/${collection.slug}`}
+                        quote={collection.title}
+                      >
+                        <FacebookIcon size={36} round={true} />
+                      </FacebookShareButton>
+                    </Grid>
+                    <Grid item>
+                      <TwitterShareButton
+                        url={`https://modularhistory.com/collections/${collection.slug}`}
+                        title={collection.title}
+                      >
+                        <TwitterIcon size={36} round={true} />
+                      </TwitterShareButton>
+                    </Grid>
+                    <Grid item>
+                      <EmailShareButton
+                        url={`https://modularhistory.com/collections/${collection.slug}`}
+                        subject={collection.title}
+                      >
+                        <EmailIcon size={36} round={true} />
+                      </EmailShareButton>
+                    </Grid>
```

Comment:
Looks like this code is duplicated between mobile and desktop view components. It can be moved to a common child component.

Reply:
Resolved in a new commit!